crates/wdk-build/src/utils.rs

```diff
@@ -374,39 +398,154 @@ fn read_registry_key_string_value(
+            assert_eq!(BasicVersion::parse("01.2"), Some(BasicVersion::new(1, 2)));
+        }
+
+        #[test]
+        fn test_parse_invalid_versions() {
+            // No dot
+            assert_eq!(BasicVersion::parse("1"), None);
+            assert_eq!(BasicVersion::parse("123"), None);
+
+            // Multiple dots
+            assert_eq!(BasicVersion::parse("1.2.3"), None);
+            assert_eq!(BasicVersion::parse("1.2.3.4"), None);
+
+            // Empty string
+            assert_eq!(BasicVersion::parse(""), None);
+
+            // Only dot
+            assert_eq!(BasicVersion::parse("."), None);
+
+            // Missing major or minor
+            assert_eq!(BasicVersion::parse(".2"), None);
+            assert_eq!(BasicVersion::parse("1."), None);
+
+            // Non-numeric values
+            assert_eq!(BasicVersion::parse("a.b"), None);
+            assert_eq!(BasicVersion::parse("1.b"), None);
+            assert_eq!(BasicVersion::parse("a.2"), None);
+            assert_eq!(BasicVersion::parse("1.2a"), None);
+            assert_eq!(BasicVersion::parse("1a.2"), None);
+
+            // Negative numbers
+            assert_eq!(BasicVersion::parse("-1.2"), None);
+            assert_eq!(BasicVersion::parse("1.-2"), None);
+            assert_eq!(BasicVersion::parse("-1.-2"), None);
+
+            // Whitespace
+            assert_eq!(BasicVersion::parse(" 1.2"), None);
+            assert_eq!(BasicVersion::parse("1.2 "), None);
+            assert_eq!(BasicVersion::parse("1 .2"), None);
+            assert_eq!(BasicVersion::parse("1. 2"), None);
+        }
+
+        #[test]
+        fn test_version_ordering() {
+            let v1_0 = BasicVersion::new(1, 0);
+            let v1_1 = BasicVersion::new(1, 1);
+            let v1_999 = BasicVersion::new(1, 999);
+            let v2_0 = BasicVersion::new(2, 0);
+            let v2_1 = BasicVersion::new(2, 1);
+
+            // Test ordering
+            assert!(v1_0 < v1_1);
+            assert!(v1_1 < v1_999);
+            assert!(v1_999 < v2_0);
+            assert!(v2_0 < v2_1);
+        }
+
+        #[test]
+        fn test_equality() {
+            let v1 = BasicVersion::new(1, 2);
+            let v2 = BasicVersion::new(1, 2);
+            let v3 = BasicVersion::new(1, 3);
+
+            assert_eq!(v1, v2);
+            assert_ne!(v1, v3);
+        }
+
+        #[test]
+        fn test_clone_and_copy() {
+            let v1 = BasicVersion::new(1, 2);
+            let v2 = v1;
+            let v3 = v1.clone();
+
+            assert_eq!(v1, v2);
+            assert_eq!(v1, v3);
+            assert_eq!(v2, v3);
+        }
+
+        #[test]
+        fn test_debug_formatting() {
+            let version = BasicVersion::new(1, 2);
+            let debug_str = format!("{:?}", version);
+            assert!(debug_str.contains("BasicVersion"));
+            assert!(debug_str.contains("major: 1"));
+            assert!(debug_str.contains("minor: 2"));
```

Comment:
```suggestion
            assert_eq!(debug_str, "BasicVersion { major: 1, minor: 2 }");
```

Instead of the `contains` checks, better to just do a single exact check against the full debug string